OptimizelySDK/Optimizely.cs

```diff
@@ -338,54 +339,60 @@ private bool ValidateInputs(string datafile, bool skipJsonValidation)
         /// <summary>
         /// Sends conversion event to Optimizely.
         /// </summary>
-        /// <param name="eventKey">Event key representing the event which needs to be recorded</param>
+        /// <param name="eventKey">Event key representing the event (must not be null, empty, or whitespace)</param>
         /// <param name="userId">ID for user</param>
         /// <param name="userAttributes">Attributes of the user</param>
         /// <param name="eventTags">eventTags array Hash representing metadata associated with the event.</param>
         public void Track(string eventKey, string userId, UserAttributes userAttributes = null,
             EventTags eventTags = null
         )
         {
-            var config = ProjectConfigManager?.GetConfig();
-
-            if (config == null)
+            if (eventKey == null || Regex.IsMatch(eventKey, @"^\s*$"))
```

Comment:
My reason for adding this earlier is because it's slightly cheaper to test the eventKey is a good string before digging into the ProjectConfig to see if the eventKey is present in the datafile since it's [required per the docs](https://docs.developers.optimizely.com/feature-experimentation/docs/track-event-csharp#:~:text=Description-,eventKey%0Arequired,-String).